### ISSUE TITLE
install: Windows: set GOPATH from installer and optionally add GOPATH\bin to PATH

### ISSUE THREAD

**Issue by gonutz** [golang/go#23025]
I ran the Go 1.9.2 Windows installers 32 bit on Windows XP and 64 bit on Windows 7.

The default `GOPATH` on Windows is now (starting with Go 1.8 I think) `%USERPROFILE%\go` but this is not set as an environment variable. It seems to just be assumed by the tools if the environment variable is not set.
The problem with that is that I cannot open `cmd` and type `cd %GOPATH%\bin` for example. This is very inconvenient and confusing to new Go users who read about `GOPATH` everywhere in the docs.

Similarly it would be a very user-friendly feature if the Windows installer asked you, after setting `GOPATH`, if it should add `%GOPATH%\bin` to your `PATH` to be able to run Go programs that they got with `go get` from `cmd` directly. This option and the explanation would be way better right there in the installer instead of having it appear on the download page as mentioned in #18583. New users might not notice the redirect (or have it disabled in the browser) or care to read it. Honestly, who reads the manual before trying things out?

Having a Windows installer is all about convenience so I think we should provide the most common tasks right in it.

This is not only for new users either. I have been using Go for years and installed it on a couple of Windows systems. After the second time I knew what I had to do but it is still tedious. Yesterday I wanted to try compiling my game on a fresh Windows, ideally I would just install Git, install Go (which would set `GOPATH` and add `%GOPATH%\bin` to `PATH`) and `go get` it. Instead I had to go through the arcane Windows menus to get through to the environment variables. (Linux is not better, every dekstop has a different hidden file and syntax to set this, but I expect Linux users to know this stuff better).

In summary my proposal is to add

- one page for setting `GOPATH`, along with a short explanation
- one page for adding `%GOPATH%\bin` to `PATH`, again with a short explanation

to the Windows installer.

**Comment by as** [comment]
```Instead I had to go through the arcane Windows menus to get through to the environment variables. (Linux is not better, every dekstop has a different hidden file and syntax to set this, but I expect Linux users to know this stuff better).```

Windows has `setx`, it sets a permanent user-wide environment variable (among 100 other weird operations). No `arcane menus` are necessary.

```
setx GOPATH C:\g
setx GOBIN C:\g\bin
...and then open a new cmd
```

**Comment by gonutz** [comment]
Which does not at all invalidate the issue but you are right, I forgot about `setx`.

**Comment by gonutz** [comment]
If anybody could point me to the code for building the installer, I would try to do it myself. I am interested in how that MSI tool works, actually. The I could create a github pull request. They should be accepted from the team by now, right?

**Comment by bradfitz** [comment]
https://github.com/golang/build/blob/master/cmd/release/releaselet.go#L220

We don't yet accept Github PRs, but we should in the next month or two. The bot to auto-convert them into Gerrit CLs is underway and starting to work.

**Comment by gonutz** [comment]
Nice, thanks very much. I will have some time for it then, that is fine with me ;-)
